contracts/SBCDepositContract.sol

```diff
@@ -374,14 +377,30 @@ contract SBCDepositContract is
             } else {
                 failedWithdrawals[numberOfFailedWithdrawals] = FailedWithdrawalRecord({
                     amount: amount,
-                    receiver: _addresses[i]
+                    receiver: _addresses[i],
+                    withdrawalIndex: nextWithdrawalIndex
                 });
+                failedWithdrawalByIndex[nextWithdrawalIndex] = numberOfFailedWithdrawals;
                 emit WithdrawalFailed(numberOfFailedWithdrawals, amount, _addresses[i]);
                 ++numberOfFailedWithdrawals;
             }
+
+            // First withdrawal is index 0
+            nextWithdrawalIndex++;
         }
     }
 
+    /**
+     * @dev Check if a block's withdrawal has been fully processed or not
+     * @param _withdrawalIndex EIP-4895 withdrawal.index property
+     */
+    function isWithdrawalProcessed(uint64 _withdrawalIndex) external view returns (bool) {
+        require(_withdrawalIndex < nextWithdrawalIndex, "withdrawal_index out-of-bounds");
+        // If there are no withdrawals failedWithdrawalByIndex returns zero, failedWithdrawals empty struct
```

Comment:
Thats a little bit wrong. In case corresponding withdrawal was succeeded initially, `failedWithdrawalByIndex[_withdrawalIndex]` will return zero, but `failedWithdrawals[0]` can correspond to some valid failed withdrawal.

Reply:
Good point, shifted the value stored in `failedWithdrawalByIndex` by one to pick up that case. Not sure if this is the cleanest approach tho. Should `failedWithdrawalIndex` start at 1 instead of 0? However since it's a proxy it complicates the init logic a bit